src/main/java/ch/njol/skript/effects/EffSpread.java

```diff
@@ -0,0 +1,114 @@
+	"spread (shuffled all players) across {_winner} and {_losers::*}",
+})
+@Since("INSERT VERSION")
+public class EffSpread extends Effect {
+
+	static {
+		Skript.registerEffect(EffSpread.class, "spread %objects% across %objects%");
+	}
+
+	private static final ClassInfoReference SINGLE_OBJECT_REFERENCE = new ClassInfoReference(DefaultClasses.OBJECT);
+
+	private Expression<?> objectsToSpread;
+	private ExpressionList<?> spreadTarget;
+
+	@Override
+	public boolean init(Expression<?>[] expressions, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		objectsToSpread = expressions[0];
+		if (objectsToSpread.isSingle()) {
+			Skript.error("You must provide more than one object to spread");
+			return false;
+		}
+		Expression<?> potentialSpreadTarget = expressions[1];
+		if (!(potentialSpreadTarget instanceof ExpressionList<?>)) {
+			Skript.error("The spread target must be a list of settable expressions");
+			return false;
+		}
+		spreadTarget = (ExpressionList<?>) potentialSpreadTarget;
+		if (!spreadTarget.getAnd()) {
+			return false;
+		}
+		if (!listChildrenCanBeSet(spreadTarget)) {
+			Skript.error("All expressions in the spread target list must be settable");
+		}
+		return true;
+	}
+
+	@Override
+	protected void execute(Event event) {
+		Object[] objectsToSpread = this.objectsToSpread.getArray(event);
+		Expression<?>[] spreadTargets = spreadTarget.getExpressions();
+		int finalObjectIndex = objectsToSpread.length - 1;
+		int objectIndex = 0;
+		for (int targetIndex = 0; targetIndex < spreadTargets.length; targetIndex++) {
+			if (objectIndex > finalObjectIndex) {
+				return;
+			}
+			Expression<?> spreadTarget = spreadTargets[targetIndex];
+			if (spreadTarget.isSingle()) {
+				spreadTarget.change(event, new Object[] { objectsToSpread[objectIndex] }, ChangeMode.SET);
+				objectIndex++;
+			} else {
+				int remainingElements = computeRemainingElements(spreadTargets, targetIndex);
+				// if there's fewer objects available than remaining, we don't want to be greedy
+				if ((objectIndex + remainingElements) > finalObjectIndex) {
+					spreadTarget.change(event, new Object[] { objectsToSpread[objectIndex] }, ChangeMode.SET);
+					objectIndex++;
+				} else {
+					// otherwise, take as much as we can leaving only enough for the remaining spread targets
+					Object[] objectsSlice = Arrays.copyOfRange(objectsToSpread, objectIndex, objectsToSpread.length - remainingElements);
+					objectIndex += objectsSlice.length;
+					spreadTarget.change(event, objectsSlice, ChangeMode.SET);
+				}
+			}
+		}
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return "spread " + objectsToSpread.toString(event, debug) + " across " + spreadTarget.toString(event, debug);
+	}
+
+	private boolean listChildrenCanBeSet(ExpressionList<?> list) {
+		for (Expression<?> child : list.getExpressions()) {
+			if (!ChangerUtils.acceptsChange(child, ChangeMode.SET, SINGLE_OBJECT_REFERENCE)) {
```

Comment:
is it safe to only check for single object? Can we not use objectsToSpread.getReturnType()?